onnxruntime/test/util/default_providers.cc

```diff
@@ -5,15 +5,15 @@
 #include "providers.h"
 #include "core/session/onnxruntime_cxx_api.h"
 #define FACTORY_PTR_HOLDER \
-  std::unique_ptr<ONNXRuntimeProviderFactoryPtr> ptr_holder_(f);
+  std::unique_ptr<ONNXRuntimeProviderFactoryInterface*> ptr_holder_(f);
```

Comment:
I would prefer having a typedef for "ONNXRuntimeProviderFactoryInterface*". Have you ever written code like:  "std::unique_ptr<T*> p" ?

Reply:
No, but we'll figure out the double pointer stuff in the next changes. It exposes your double pointers :)